types/wlr_box.c

```diff
@@ -143,3 +143,19 @@ void wlr_box_rotated_bounds(const struct wlr_box *box, float rotation,
 	dest->y = floor(fmin(y1, y2));
 	dest->height = ceil(fmax(y1, y2) - fmin(y1, y2));
 }
+
+void wlr_box_from_pixman_box32(const pixman_box32_t box, struct wlr_box *dest) {
+	*dest = (struct wlr_box){
+		.x = box.x1,
+		.y = box.y1,
+		.width = box.x2 - box.x1,
+		.height = box.y2 - box.y1,
+	};
+}
+
+const struct wlr_box WLR_BOX_MAX = {
+	.x = 0,
+	.y = 0,
```

Comment:
I think `x` and `y` need to be INT_MIN.

Reply:
Yeah, we do allow layouts to configure their outputs at negative x/y coordinates. It does make this a little tricky though.